cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -3,60 +3,96 @@ package cwl
 import wom.types.WomType
+import wom.values.WomString
 import wom.{CommandPart, RuntimeAttributes}
 
+import scala.language.postfixOps
 import scala.util.Try
 
 /**
   * @param `class` This _should_ always be "CommandLineTool," however the spec does not -er- specify this.
   */
 case class CommandLineTool private(
-                                    inputs: Array[CommandInputParameter],
-                                    outputs: Array[CommandOutputParameter],
-                                    `class`: Witness.`"CommandLineTool"`.T,
-                                    id: String,
-                                    requirements: Option[Array[Requirement]],
-                                    hints: Option[Array[CwlAny]],
-                                    label: Option[String],
-                                    doc: Option[String],
-                                    cwlVersion: Option[CwlVersion],
-                                    baseCommand: Option[BaseCommand],
-                                    arguments: Option[Array[CommandLineTool.Argument]],
-                                    stdin: Option[StringOrExpression],
-                                    stderr: Option[StringOrExpression],
-                                    stdout: Option[StringOrExpression],
-                                    successCodes: Option[Array[Int]],
-                                    temporaryFailCodes: Option[Array[Int]],
-                                    permanentFailCodes: Option[Array[Int]]) {
+                                   inputs: Array[CommandInputParameter],
+                                   outputs: Array[CommandOutputParameter],
+                                   `class`: Witness.`"CommandLineTool"`.T,
+                                   id: String,
+                                   requirements: Option[Array[Requirement]],
+                                   hints: Option[Array[Hint]],
+                                   label: Option[String],
+                                   doc: Option[String],
+                                   cwlVersion: Option[CwlVersion],
+                                   baseCommand: Option[BaseCommand],
+                                   arguments: Option[Array[CommandLineTool.Argument]],
+                                   stdin: Option[StringOrExpression],
+                                   stderr: Option[StringOrExpression],
+                                   stdout: Option[StringOrExpression],
+                                   successCodes: Option[Array[Int]],
+                                   temporaryFailCodes: Option[Array[Int]],
+                                   permanentFailCodes: Option[Array[Int]]) {
+
+  /** Builds an `Executable` directly from a `CommandLineTool` CWL with no parent workflow. */
+  def womExecutable(validator: RequirementsValidator, inputFile: Option[String] = None): Checked[Executable] = {
```

Comment:
ToL Maybe creating a new type using the Reader Monad could clean up the passing around of this validator.  I'll give it a quick shot and see if it just makes things better or worse.